### ISSUE TITLE
Global init on YYC

### ISSUE THREAD

**Issue by YellowAfterlife** [YAL-Haxe/sfgml#19]
Related bug report:

> ## [YYC: Functions "inherit" local function argument array size](https://help.yoyogames.com/hc/en-us/requests/196743)
> 
> [same in beta and stable]
> 
> Case A (scr_1): you have a script with global init code - perhaps a CRC table, or an array of strings for enum "field" names. Each function inside the script will have its __Args__/__pArgs__ allocated to that many arguments despite not needing that.
> 
> Case B (scr_2): you have a function with local functions in it - perhaps callbacks, or constructor methods, or anything else. These local functions will similarly size their argument arrays based on the container's array instead of their needs.
> 
> In my case, I had a large file with numerous helper functions and related data and was wondering why even the tiniest functions were somewhat slow to call before pinpointing the issue to this.

This can result in code like this:
```cpp
YYRValue& gml_Script_scr_1( CInstance* pSelf, CInstance* pOther, YYRValue& _result, int _count,  YYRValue** _args  )
{
YY_STACKTRACE_FUNC_ENTRY( "gml_Script_scr_1", 0 );
YYGML_array_set_owner( (int64)(intptr_t)pSelf );
YYRValue* __pArgs__[128];
YYRValue __Args__[128];

_result.kind = VALUE_UNDEFINED; _result.ptr = NULL;

YY_STACKTRACE_LINE(3);
YYGML_show_debug_message(YYRValue(g_pString26_0F24E86A));
return _result;
}
```
To get around this, sfgml will for a time wrap type init (enum name arrays being the worst offender here) in a `(function() {...})()` call.

**Comment by YellowAfterlife** [comment]
[a_bug.yyz.zip](https://github.com/YAL-Haxe/sfgml/files/8501021/a_bug.yyz.zip)

**Comment by YellowAfterlife** [comment]
Checking back on the bugtracker, apparently this was fixed around 2022.6 release, but I might as well keep the closure in case anything else breaks.
